### PR TITLE
Update cardano-api-8.39.0.0

### PR DESCRIPTION
**Pull request by carbolymer**
# Changelog

```yaml
- description: |
    Update [cardano-api-8.39.0.0](https://github.com/IntersectMBO/cardano-api/blob/main/cardano-api/CHANGELOG.md#83900)
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
   - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

This PR contains indices bump and redundant imports removal, because of how cardano-api exports things now.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Self-reviewed the diff

### CODE REVIEW

File: cardano-cli/test/cardano-cli-golden/files/golden/governance/hash/foo.script.hash [hunk 1]
```diff
@@ -1 +1 @@
-2a7a40032759083907e43e145b770b467cfc0ddbc3c1266bc2c01b40
\ No newline at end of file
+7dde1be4d1a2f617ac343dd7ef0388cf497c19ba8dab653a5b9eb1e2
```
Comment (smelc) [inline cardano-cli/test/cardano-cli-golden/files/golden/governance/hash/foo.script.hash:1]: What's the reason for this hash change?

File: cardano-cli/test/cardano-cli-test-lib/Test/Cardano/CLI/Util.hs [hunk 2]
```diff
@@ -15,14 +15,10 @@ module Test.Cardano.CLI.Util
...
 
 import           Cardano.Api
-import           Cardano.Api.Pretty
 
 import           Cardano.CLI.Read
 
 import           Control.Monad.Catch
-import           Control.Monad.IO.Class (MonadIO (..))
-import           Control.Monad.Trans.Class (lift)
-import           Control.Monad.Trans.Except (runExceptT)
```
Comment (smelc) [inline cardano-cli/test/cardano-cli-test-lib/Test/Cardano/CLI/Util.hs:25]: Why are all these imports being removed? Were they useless in the first place?

**Comment by carbolymer** [comment]
Hydra windows build is failing because of haskell.nix + blst on windows issues: https://input-output-rnd.slack.com/archives/CG1FBSDMM/p1708462370264619

File: cardano-cli/test/cardano-cli-test-lib/Test/Cardano/CLI/Util.hs (hunk 2 above)
Comment (carbolymer) [reply cardano-cli/test/cardano-cli-test-lib/Test/Cardano/CLI/Util.hs:25]: They are now reexported in `Cardano.Api.Monad.Error`, which is exported in `Cardano.Api`. Context: https://github.com/IntersectMBO/cardano-api/pull/453#discussion_r1494613609

File: cardano-cli/test/cardano-cli-golden/files/golden/governance/hash/foo.script.hash (hunk 1 above)
Comment (carbolymer) [reply cardano-cli/test/cardano-cli-golden/files/golden/governance/hash/foo.script.hash:1]: I think that's an expected result from: https://github.com/IntersectMBO/cardano-api/pull/452

File: cardano-cli/test/cardano-cli-test-lib/Test/Cardano/CLI/Util.hs (hunk 2 above)
Comment (smelc) [reply cardano-cli/test/cardano-cli-test-lib/Test/Cardano/CLI/Util.hs:25]: @carbolymer> Thanks for the pointer :+1:

**Review by smelc — APPROVED** [review]

**Review by smelc — APPROVED** [review]

**Comment by palas** [comment]
Sorry, it seems I did the same exact thing you did in https://github.com/IntersectMBO/cardano-cli/pull/558 (really wanted to get it merged). I misunderstood something and it didn't occur to me to check whether the imports had been fixed in another PR until after doing it myself 🤦.
Anyway, the way you fixed `cardano-cli/src/Cardano/CLI/Pretty.hs` is much nicer, I didn't know you could do that.

**Comment by carbolymer** [comment]
@palas no worries, thanks for doing that! I've just updated the description of #558.

**Comment by carbolymer** [comment]
Superceded by https://github.com/IntersectMBO/cardano-cli/pull/558

**Pull request closed without merging**